src/tls.c

```diff
@@ -11043,7 +11043,7 @@ int TLSX_Parse(WOLFSSL* ssl, const byte* input, word16 length, byte msgType,
         ato16(input + offset, &size);
         offset += OPAQUE16_LEN;
 
-        if (offset + size > length)
+        if (length - offset < size)
```

Comment:
What case is this detecting that the previous one did not?

Reply:
`size` is the value read out of the buffer. If the unsigned value `size` is sufficiently large, when added to the unsigned value `offset`, it could overflow (or wrap around) to a number smaller than `length`:
size = 0xFFFF
offset = 0x10
length = 0x30

The wrap around forces `size + offset` to be 0xF

By checking the "known good" values of length and offset against the unknown value of size, the overflow is prevented.
https://stackoverflow.com/questions/16056758/c-c-unsigned-integer-overflow

Reply:
That's true, thanks.